Add animation transitions and revamp code style

AnotherMedo opened the pull request:
This commit makes substantial changes to the repository and will serve as a setup for the remainder of the project development:
- Abstracted multiple composables in `app/src/main/java/com/argym/quickfix/ui/elements` to be completed/used in future composables
- Added transitions between authentications screens for smoother more pleasing and more professional viewing experience
- Added the font specified in the Figma (i.e. `Inter`) in its `ExtraBold` and `ExtraBoldItalic` versions
- Abstracted the color scheme and removed the `LoD` parameter of all composables and replacing it with its intended counterpart (c.f. `Theme.kt`)
- Abstracted text styles

This PR aims at making development less confusing and cumbersome and aid in faster and more efficient development with more reusable components. It also aims to promote ***modular development*** for future PRs 🚀 😄 


Edit: This PR ended up being more of a "save the repository" kind of thing with adding tests and fixing everything that was wrong with the repository before moving on with the project.

JitterKiller commented:
Make sure to format your code with KTFMT, as it's currently failing the CI checks. Other than that, great work!

JitterKiller reviewed (changes requested):
Make sure to format your code with KTFMT, as it's currently failing the CI checks. Other than that, great work!

AnotherMedo commented:
> Make sure to format your code with KTFMT, as it's currently failing the CI checks. Other than that, great work!

These are formatting issues that were pushed in #32 they do not come from my code 😅. I must've forgot to pull before making my changes I will look into it

daferya commented on app/src/main/java/com/arygm/quickfix/ui/Announcement.kt:
```diff
@@ -26,56 +26,46 @@ import com.arygm.quickfix.ui.navigation.Route
 
-    val color1 = if (LoD) Color(0xFFF16138) else Color(0xFF633040)
-    val backgroundColor = if (LoD) Color.White else Color(0xFF282828)
+  val color1 = if (LoD) Color(0xFFF16138) else Color(0xFF633040)
+  val backgroundColor = if (LoD) Color.White else Color(0xFF282828)
```
The color1 and backgroundColor variables are declared twice. Move them outside the CalendarScreen function to define them globally, or place them in a separate theme configuration file if they are reused elsewhere in the app

AnotherMedo replied:
done :)

daferya reviewed (commented):
Since Scaffold is also duplicated, you can create a wrapper function that takes content and topBar as parameters. This centralizes the layout logic and prevents repetitive code
The color1 and backgroundColor variables are declared twice. Move them outside the CalendarScreen function to define them globally, or place them in a separate theme configuration file if they are reused elsewhere in the app.
To meet the coverage threshold, tests should be added for the UI components and navigation logic,

mohagpt commented on app/src/main/java/com/arygm/quickfix/ui/Announcement.kt:
```diff
@@ -26,56 +26,46 @@ import com.arygm.quickfix.ui.navigation.Route
+        Surface(
+            shape = RoundedCornerShape(bottomStart = 16.dp, bottomEnd = 16.dp),
+            modifier = Modifier.fillMaxWidth()) {
+              TopAppBar(
+                  title = {
+                    Box(modifier = Modifier.fillMaxWidth()) {
+                      Text(
+                          text = "Announcement",
+                          color = backgroundColor,
+                          modifier = Modifier.align(Alignment.Center))
+                    }
+                  },
+                  colors = TopAppBarDefaults.topAppBarColors(containerColor = color1))
             }
```
The top app bar beeing present in a lot of screens can be abstracted in another file.

mohagpt commented on app/src/main/java/com/arygm/quickfix/ui/Calendar.kt:
```diff
@@ -26,56 +26,45 @@ import com.arygm.quickfix.ui.navigation.Route
+        Surface(
+            shape = RoundedCornerShape(bottomStart = 16.dp, bottomEnd = 16.dp),
+            modifier = Modifier.fillMaxWidth()) {
+              TopAppBar(
+                  title = {
+                    Box(modifier = Modifier.fillMaxWidth()) {
+                      Text(
+                          text = "Calendar",
+                          color = backgroundColor,
+                          modifier = Modifier.align(Alignment.Center))
+                    }
+                  },
+                  colors = TopAppBarDefaults.topAppBarColors(containerColor = color1))
             }
```
Same here for the top app bar

mohagpt reviewed (approved):
> The only thing i can add is to abstract the  top app Bar in a new file, as it has been done for other components.

Sebbayra commented on app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixAnimatedBox.kt:
```diff
@@ -0,0 +1,33 @@
+    color: Color = colorScheme.primary,
+) {
+  Box(
+      modifier =
+          Modifier.requiredSize(size)
+              .offset(x = xOffset, y = yOffset)
+              .graphicsLayer(rotationZ = rotation)
+              .background(color)
+              .zIndex(1f))
```
Add a test tag for UI testing.

Sebbayra commented on app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixCheckBoxRow.kt:
```diff
@@ -0,0 +1,50 @@
+    colorScheme: ColorScheme = MaterialTheme.colorScheme
+) {
+  Row(modifier = Modifier.fillMaxWidth(), verticalAlignment = Alignment.CenterVertically) {
+    Checkbox(
+        checked = checked,
+        onCheckedChange = onCheckedChange,
+        colors =
+            CheckboxDefaults.colors(
+                checkedColor = colorScheme.primary,
+                uncheckedColor = colorScheme.tertiary,
+                checkmarkColor = Color.Transparent),
+        modifier = Modifier.size(24.dp))
```
Add checkbox test tag.

Sebbayra commented on app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixCheckBoxRow.kt:
```diff
@@ -0,0 +1,50 @@
+    Spacer(modifier = Modifier.width(10.dp))
+    Text(label, style = MaterialTheme.typography.labelSmall, color = Color(0xFFC0C0C0))
+    Spacer(modifier = Modifier.width(4.dp))
+    Text(
+        text = underlinedText,
+        style = MaterialTheme.typography.labelSmall,
+        color = colorScheme.primary,
+        textDecoration = TextDecoration.Underline,
+        modifier = Modifier.clickable(onClick = onUnderlinedTextClick))
```
Test tag here as well

Sebbayra commented on app/src/main/java/com/arygm/quickfix/ui/elements/QuickFixCheckBoxRow.kt:
```diff
@@ -0,0 +1,50 @@
+                checkmarkColor = Color.Transparent),
+        modifier = Modifier.size(24.dp))
+    Spacer(modifier = Modifier.width(10.dp))
+    Text(label, style = MaterialTheme.typography.labelSmall, color = Color(0xFFC0C0C0))
```
Test tag.

Sebbayra reviewed (changes requested):
Overall excellent PR, the only thing i would add is the couple test tags to generate more coverage, i added them in the sub branch of testing of this branch.
We should also add the valid mail format for the login Screen.
One thing i also fixed in the sub-branch is the bug where we can press the register button in the Password screen if we just enter a valid password and keep the repeat password empty, so just approve my testing branch and this should do the job.

AnotherMedo commented:
> Overall excellent PR, the only thing i would add is the couple test tags to generate more coverage, i added them in the sub branch of testing of this branch. We should also add the valid mail format for the login Screen. One thing i also fixed in the sub-branch is the bug where we can press the register button in the Password screen if we just enter a valid password and keep the repeat password empty, so just approve my testing branch and this should do the job.

Solved in #40

JitterKiller reviewed (dismissed):
Overall Great PR, this will really improve our code quality and reduce duplication code.
Great work!

AnotherMedo commented:
> Overall excellent PR, the only thing i would add is the couple test tags to generate more coverage, i added them in the sub branch of testing of this branch. We should also add the valid mail format for the login Screen. One thing i also fixed in the sub-branch is the bug where we can press the register button in the Password screen if we just enter a valid password and keep the repeat password empty, so just approve my testing branch and this should do the job.

Solved in #41

Sebbayra reviewed (approved):
Great, everything is ready to be merged, great work.

JitterKiller reviewed (approved):
Great catch for the CI implementation, Sonar is now working and consistent with jaCoCo test reports.
Concerning the UI side, everything looks good to me!
This PR is now ready to be merged exeptionally (even tho we're not at 80% code coverage)

JitterKiller reviewed (approved):
Great catch for the CI implementation, Sonar is now working and consistent with jaCoCo test reports.
Concerning the UI side, everything looks good to me!
This PR is now ready to be merged exeptionally (even tho we're not at 80% code coverage)

Sebbayra reviewed (approved):
Looks, amazing, we can finally merge, thank you goat mehdi.
